InstallDatabases.bat

```diff
@@ -1,1263 +1,1308 @@
+echo ^| with no external internet access.                                           ^|
+echo ^|                                                                             ^|
+echo ^| Using these settings will leave you server open to exploits,                ^|
+echo ^| security breaches... or much much worse.                                    ^|
+echo ^|_____________________________________________________________________________^|%colReset%
+echo.
+echo.
+set /p svr=%colYellowBold% What is your MySQL host name?           %colReset%DEFAULT: [%colWhiteBold%%svr%%colReset%] :
+if %svr%. == . set svr=localhost
+set /p user=%colYellowBold% What is your MySQL user name?                %colReset%DEFAULT: [%colWhiteBold%%user%%colReset%] :
+if %user%. == . set user=root
+set /p pass=%colYellowBold% What is your MySQL password?               %colReset%DEFAULT: [%colWhiteBold%%pass%%colReset%] :
+if %pass%. == . set pass=mangos
+set /p port=%colYellowBold% What is your MySQL port?                     %colReset%DEFAULT: [%colWhiteBold%%port%%colReset%] :
+if %port%. == . set port=3306
+
+set showChar=0
+if %createcharDB% == YES set showChar=1
+if %loadcharDB% == YES set showChar=1
+if %CDBUpdate% == YES set showChar=1
+
+if %showChar% == 1 set /p cdb=%colYellowBold% What is your Character database name?  %colReset%DEFAULT: [%colWhiteBold%%cdb%%colReset%] :
+if %cdb%. == . set cdb=%cdborig%
+
+set showWorld=0
+if %createworldDB% == YES set showWorld=1
+if %loadworldDB% == YES set showWorld=1
+if %WDBUpdate% == YES set showWorld=1
+if %showWorld% == 1 set /p wdb=%colYellowBold% What is your World database name?         %colReset%DEFAULT: [%colWhiteBold%%wdb%%colReset%] :
+if %wdb%. == . set wdb=%wdborig%
+
+set showRealm=0
+if %createrealmDB% == YES set showRealm=1
+if %loadrealmDB% == YES set showRealm=1
+if %addrealmentry% == YES set showRealm=1
+if %CDBUpdate% == YES set showRealm=1
+
+if %showRealm% == 1 set /p rdb=%colYellowBold% What is your Realm database name?          %colReset%DEFAULT: [%colWhiteBold%%rdb%%colReset%] :
+if %rdb%. == . set rdb=%rdborig%
+
+REM If create user is set to yes, ask for the details here
+if %createMangosUser% == NO goto WorldDB:
+
+if %createMangosUser% == YES set /p newuser=%colYellowBold% New MySQL user name?                       %colReset%DEFAULT: [%colWhiteBold%%newuser%%colReset%] :
+if %newuser%. == . set newuser=mangos
+if %createMangosUser% == YES set /p newpass=%colYellowBold% New MySQL user password?                   %colReset%DEFAULT: [%colWhiteBold%%newpass%%colReset%] :
+if %newpass%. == . set newpass=mangos
+
+:AfterStep1
+if "%mysqlAuth%" == "" set mysqlAuth=-h %svr% --user=%user% --password=%pass% --port=%port%
+set mysqlString="%mysqlDir:"=%mysql.exe" %mysqlAuth% -q -s
+if "%DEBUG%" == "true" set mysqlString=echo %mysqlString%
+if "%DEBUG%" == "TRUE" set mysqlString=echo %mysqlString%
+
+if %createMangosUser% == YES if %newpass% == mangos set defaultsused=YES

```

Comment:
Is line 497 correct, the syntax looks a little 'off'

Reply:
Those are nested terse conditions, omitting 2 pairs of optional parens.